### PR TITLE
Improving login

### PR DESCRIPTION
**Pull request by maximeroucher**
- Adding the possibility to pass the activation with a query parameter `code`, which will add the activation code field on UI
- Adding a password confirmation
- Fixing animation issue due to the routing addition
- Fixing the password strength checker bar UI

### CODE REVIEW

File: lib/login/ui/sign_in.dart [hunk 1]
```diff
@@ -17,113 +20,129 @@ class SignIn extends HookConsumerWidget {
...
+          autovalidateMode: AutovalidateMode.onUserInteraction,
+          child: Padding(
+            padding: const EdgeInsets.all(20.0),
+            child: Column(
+              children: [
+                Expanded(
+                  flex: 3,
+                  child: Align(
+                    alignment: Alignment.centerLeft,
+                    child: Text("MyECL",
```
Comment (Rotheem) [inline lib/login/ui/sign_in.dart:42]: Use text constant to match your own requirement please

File: lib/login/ui/sign_in.dart [hunk 2]
```diff
@@ -17,113 +20,129 @@ class SignIn extends HookConsumerWidget {
...
-                          Container(
-                              height: 40,
-                              alignment: Alignment.centerLeft,
-                              child: InkWell(
-                                splashColor:
-                                    const Color.fromRGBO(255, 255, 255, 1),
-                                onTap: () {
-                                  QR.to(LoginRouter.createAccount);
+                              SignInUpBar(
+                                isLoading: ref.watch(loadingrovider).when(
```
Comment (Rotheem) [inline lib/login/ui/sign_in.dart:64]: LoadingProvider would be better

**Review by Rotheem — CHANGES_REQUESTED** [review]

File: lib/login/ui/sign_in.dart (hunk 2 above)
Comment (maximeroucher) [reply lib/login/ui/sign_in.dart:64]: Already renamed in Refactoring PR 😉

File: lib/login/ui/sign_in.dart (hunk 1 above)
Comment (maximeroucher) [reply lib/login/ui/sign_in.dart:42]: Already done in the Refactoring PR : https://github.com/aeecleclair/Titan/pull/91 😉

**Review by Rotheem — APPROVED** [review]

**Pull request merged**